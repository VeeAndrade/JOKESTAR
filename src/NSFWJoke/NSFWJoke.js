import React from 'react';
import './NSFWJoke.scss';
import Joke from '../Joke/Joke';
import { connect } from 'react-redux';

export const NSFWJoke = (props) => {
  return (
    <section className='nsfw-joke-container'>
      <section className='joke-section'>
        <Joke joke={props.joke}/>
      </section>
      <section className='buttons-section'>
        <button className='favorite-btn' onClick={props.favoriteJoke}>Favorite</button>
        <button className='next-joke-btn' onClick={props.getNextJoke}>Next Joke</button>
      </section>
    </section>)
}

export const mapStateToProps = state => ({
  joke: state.joke
})

export default connect(mapStateToProps)(NSFWJoke);