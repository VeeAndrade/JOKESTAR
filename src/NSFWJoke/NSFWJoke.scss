$faded-purple: rgba(74, 21, 160, 0.452);
$faded-blue: rgba(18, 125, 247, 0.452);
$bright-green: rgb(0, 255, 136);

.nsfw-joke-container {
  height: 80%;
  width: 95%;
}

.buttons-section {
  height: 5rem;
  display: flex;
  justify-content: space-around;

}

.next-joke-btn,
.favorite-btn {
  width: 35%;
  height: 2.5rem;
  border: solid 1px $bright-green;
  background-color: transparent;
  color: $bright-green;
  font-size: 1.2em;
  border-radius: 5px;
  font-weight: bold;

    &:hover {
      background-color: $bright-green;
      color: rebeccapurple;
  }
}