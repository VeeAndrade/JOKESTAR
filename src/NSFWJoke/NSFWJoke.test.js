import React from 'react';
import { NSFWJoke } from './NSFWJoke';
import { shallow } from 'enzyme';

describe('NSFWJoke', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallow( <NSFWJoke 
    joke={{id: 42, setup: 'how much does a polarbear weigh?', delivery: 'enough to bresk the ice'}}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
}); 