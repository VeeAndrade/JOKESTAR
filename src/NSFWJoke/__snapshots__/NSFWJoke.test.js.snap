// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NSFWJoke should match the snapshot 1`] = `
<section
  className="nsfw-joke-container"
>
  <section
    className="joke-section"
  >
    <Joke
      joke={
        Object {
          "delivery": "enough to bresk the ice",
          "id": 42,
          "setup": "how much does a polarbear weigh?",
        }
      }
    />
  </section>
  <section
    className="buttons-section"
  >
    <button
      className="favorite-btn"
    >
      Favorite
    </button>
    <button
      className="next-joke-btn"
    >
      Next Joke
    </button>
  </section>
</section>
`;
